core/common/fd_limit.hpp

```diff
@@ -8,7 +8,10 @@
 
 #include <cstdlib>
 
+struct rlimit;
+
 namespace kagome::common {
+  bool getFdLimit(rlimit &r);
```

Comment:
Better hide `rlimit` (`.rlim_max` is not used anyway).  
No need for `#include <sys/resource.h>` too.

```suggestion
  std::optional<size_t> getFdLimit();
```